include/my_sys.h

```diff
@@ -174,6 +174,8 @@ extern void my_free(void *ptr);
 extern void *my_memdup(PSI_memory_key key, const void *from,size_t length,myf MyFlags);
 extern char *my_strdup(PSI_memory_key key, const char *from,myf MyFlags);
 extern char *my_strndup(PSI_memory_key key, const char *from, size_t length, myf MyFlags);
+extern void *aligned_malloc(size_t size, size_t align);
+extern void aligned_free(void *ptr);
```

Comment:
Instead of replacing the InnoDB inline wrappers with non-inlined ones, what if we outside Windows used the ISO/IEC 9899:2011 (C11) standard library function `aligned_alloc()`, whose interface matches the Microsoft `_aligned_malloc()`? Something like this:
```c
#ifdef _WIN32
# define aligned_malloc(size, align) _aligned_malloc(size, align)
# define aligned_free(ptr) _aligned_free(ptr)
#else
# define aligned_malloc(size, align) aligned_alloc(size, align)
# define aligned_free(ptr) free(ptr)
#endif
```
Note: Unlike in the previous definition, the C11 standard specifies that `size` must be an integer multiple of `align`. Some code might have to be adjusted accordingly.

Also, while doing this, I think that we should remove all references to `HAVE_MEMALIGN` and make `performance_schema` use this common interface. There should be no need to use `posix_memalign()` or `memalign()`.

I am not entirely sure whether we can switch to C11 here. I would assume that it is technically possible. According to a quick check on https://godbolt.org the function `aligned_alloc()` is available in `<stdlib.h>` already in GCC 4.8.5, which is the oldest compiler that we support.

Reply:
`aligned_alloc()` is defined in ISO/IEC 9899:2011 a.k.a. C11. If it works in all environments that our CI systems include, I think that we should replace the nonstandard alternatives. Only on Microsoft Windows, nonstandard functions will have to be used.

Starting with MariaDB Server 10.4, ISO/IEC 14882:2011 a.k.a. C++11 is a build dependency, and GCC 4.8.5 is the oldest compiler that supports the subset that we require. It appears to support this part of C11 as well.

I plan to test my suggestion to check the portability. By default, pull requests are only built and tested on a small number of platforms.

Reply:
I learned that AddressSanitizer (at least in GCC 11.3.0) is enforcing the parameters of `aligned_alloc()`; I initially got the parameters the wrong way around:
```
==49198==ERROR: AddressSanitizer: invalid alignment requested in aligned_alloc: 2368, alignment must be a power of two and the requested size 0x40 must be a multiple of alignment (thread T0)
```

Reply:
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=252859 and a few other not too references suggest that instead of the C11 standard `aligned_alloc()`, `posix_memalign()` should still be used on FreeBSD and macOS.

Reply:
95f5ebdcd416761c4e295ccfaa941c243dedfe91 is my take at this.

Reply:
https://gitlab.com/drobilla/serd/-/commit/c4c4ec510dbeff61982e4aee7d1b379539319cd9 disables the use of `aligned_alloc` with the motivation that executables that were built on a newer macOS would fail to run on an older macOS. That does not look like something that we should care about, but if really needed, we could apply some hack to disable the use of `aligned_alloc` on macOS.